x-pack/plugin/logstash/src/main/java/org/elasticsearch/xpack/logstash/rest/RestPutPipelineAction.java

```diff
@@ -47,21 +47,16 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             // parse pipeline for validation
             Pipeline.PARSER.apply(parser, id);
         }
-
-        return restChannel -> {
-            final String content = request.content().utf8ToString();
-            client.execute(
-                PutPipelineAction.INSTANCE,
-                new PutPipelineRequest(id, content, request.getXContentType()),
-                new RestActionListener<>(restChannel) {
-                    @Override
-                    protected void processResponse(PutPipelineResponse putPipelineResponse) throws Exception {
-                        channel.sendResponse(
-                            new RestResponse(putPipelineResponse.status(), XContentType.JSON.mediaType(), BytesArray.EMPTY)
-                        );
-                    }
+        final String content = request.releasableContent().utf8ToString();
```

Comment:
Does this need to happen outside the final lambda now or could we move it back to where it was before?